corehq/apps/geospatial/static/geospatial/js/models.js

```diff
@@ -64,7 +64,17 @@ hqDefine('geospatial/js/models', [
             return gettext("Case");
         };
 
+        self.updateCheckbox = function () {
+            const checkbox = $(`#${self.selectCssId}`);
+            if (!checkbox) {
+                return;
+            }
+            checkbox.prop('checked', self.isSelected());
+        };
+
         self.isSelected.subscribe(function () {
+            // Popup might be open when value changes, so make sure checkbox shows correct value
```

Comment:
nit: maybe include the info about `knockout loses binding` in the comment

Reply:
That's a good idea

Reply:
[1c28fde](https://github.com/dimagi/commcare-hq/pull/34709/commits/1c28fdeca8f6b48fadeadba949c331b8001a64a5)